lib/internal/streams/web.js

```diff
@@ -0,0 +1,4489 @@
+      return PromiseReject(error);
+    }
+  }
+
+  /**
+   * @returns {ReadableStream[]}
+   */
+  tee() {
+    if (!isReadableStream(this))
+      throw new ERR_INVALID_THIS('ReadableStream');
+    return readableStreamTee(this, false);
+  }
+
+  // The values() method is not defined in the streams standard
+  // spec but it is called out in the section on async iteration.
+  // So... that's weird.
+
+  /**
+   * @param {{
+   *   preventCancel? : boolean,
+   * }} [options]
+   * @returns {AsyncIterable}
+   */
+  values(options = {}) {
+    if (!isReadableStream(this))
+      throw new ERR_INVALID_THIS('ReadableStream');
+    validateObject(options, 'options');
+    const {
+      preventCancel = false,
+    } = options;
+
+    const reader = new ReadableStreamDefaultReader(this);
+    let done = false;
+    let started = false;
+    let current;
+
+    function nextSteps() {
+      if (done)
+        return PromiseResolve({ done: true, value: undefined });
+
+      if (reader[kState].stream === undefined) {
+        return PromiseReject(
+          new ERR_INVALID_STATE.TypeError(
+            'The reader is not bound to a ReadableStream'));
+      }
+      const promise = createDeferredPromise();
+
+      readableStreamDefaultReaderRead(reader, {
+        [kChunk](chunk) {
+          current = undefined;
+          promise.resolve({ value: chunk, done: false });
+        },
+        [kClose]() {
+          current = undefined;
+          done = true;
+          readableStreamReaderGenericRelease(reader);
+          promise.resolve({ done: true, value: undefined });
+        },
+        [kError](error) {
+          current = undefined;
+          done = true;
+          readableStreamReaderGenericRelease(reader);
+          promise.reject(error);
+        }
+      });
+      return promise.promise;
+    }
+
+    function returnSteps(value) {
+      if (done)
+        return PromiseResolve({ done: true, value });
+      done = true;
+
+      if (reader[kState].stream === undefined) {
+        return PromiseReject(
+          new ERR_INVALID_STATE.TypeError(
+            'The reader is not bound to a ReadableStream'));
+      }
+      assert(!reader[kState].readRequests.length);
+      if (!preventCancel) {
+        const result = readableStreamReaderGenericCancel(reader, value);
+        readableStreamReaderGenericRelease(reader);
+        return PromisePrototypeThen(
+          result,
+          () => { return { done: true, value }; });
+      }
+
+      readableStreamReaderGenericRelease(reader);
+      return PromiseResolve({ done: true, value });
+    }
+
+    return ObjectSetPrototypeOf({
+      next() {
+        // If this is the first read, delay by one microtask
+        // to ensure that the controller has had an opportunity
+        // to properly start and perform the initial pull.
+        // TODO(@jasnell): The spec doesn't call this out so
+        // need to investigate if it's a bug in our impl or
+        // the spec.
```

Comment:
Interesting. I ran into [a similar oddity](https://github.com/MattiasBuelens/web-streams-polyfill/commit/bcc1a1b01133658bcd4554a3b46d4bfd47281846), but I never investigated further. 🤔

Reply:
Yeah I'd spotted that in your impl. Once I'm passing the rest of the WPTs I'll be revisiting.